tests/test_primitives/test_ite.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+import magma as m
+
+
+def test_adc_signed_unsigned():
+    Data = m.Bits[16]
+    UData = m.UInt[16]
+    SData = m.SInt[16]
+
+    with pytest.raises(TypeError) as e:
```

Comment:
Can we distill the test(s) for what we want to check? In particular it might be easier to read and maintain if the test just does the minimal thing, i.e. not wrapped in `combinational`. Perhaps one test to check (a) basic sint-uint wiring (see comment above), and one test for (b) mux'ing those.

Reply:
Sure, the sint-uint wiring is the underlying problem, so we need to fix that.  then there's the fact that ite (used by combinational/sequential) isn't using the mux operator which defines the type checking rules for inference (so we can change it from using combinational to just calling ite directly)